File: app/src/main/java/com/wealoha/social/beans/Image.java
package com.wealoha.social.beans;

import android.os.Parcel;
import android.os.Parcelable;

public class Image implements Parcelable {

	public String id;
	public String type;
	public int width;
	public int height;
	public String url;
	public String urlPatternWidth;
	public String urlPatternWidthHeight;

	public String path;
	public String mimeType;

	public Image() {
	}

	public Image(Parcel source) {
		id = source.readString();
		type = source.readString();
		width = source.readInt();
		height = source.readInt();
		url = source.readString();
		urlPatternWidth = source.readString();
		urlPatternWidthHeight = source.readString();
		path = source.readString();
		mimeType = source.readString();
	}

	public static final Parcelable.Creator<Image> CREATOR = new Creator<Image>() {

		@Override
		public Image createFromParcel(Parcel source) {
			return new Image(source);
		}

		@Override
		public Image[] newArray(int size) {
			return new Image[size];
		}
	};

	@Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeString(id);
		dest.writeString(type);
		dest.writeInt(width);
		dest.writeInt(height);
		dest.writeString(url);
		dest.writeString(urlPatternWidth);
		dest.writeString(urlPatternWidthHeight);
		dest.writeString(path);
		dest.writeString(mimeType);
	}

	@Override
	public int describeContents() {
		return 0;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public int getWidth() {
		return width;
	}

	public void setWidth(int width) {
		this.width = width;
	}

	public int getHeight() {
		return height;
	}

	public void setHeight(int height) {
		this.height = height;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public String getUrlPatternWidth() {
		return urlPatternWidth;
	}

	public void setUrlPatternWidth(String urlPatternWidth) {
		this.urlPatternWidth = urlPatternWidth;
	}

	public String getUrlPatternWidthHeight() {
		return urlPatternWidthHeight;
	}

	public void setUrlPatternWidthHeight(String urlPatternWidthHeight) {
		this.urlPatternWidthHeight = urlPatternWidthHeight;
	}

	public String getPath() {
		return path;
	}

	public void setPath(String path) {
		this.path = path;
	}

	public String getMimeType() {
		return mimeType;
	}

	public void setMimeType(String mimeType) {
		this.mimeType = mimeType;
	}

}

File: app/src/main/java/com/wealoha/social/beans/Image.kt
package com.wealoha.social.beans

import android.os.Parcelable
import com.wealoha.social.utils.Debug
import com.wealoha.social.utils.Dimens
import com.wealoha.social.utils.ImageUtil
import kotlinx.android.parcel.Parcelize
import org.apache.commons.lang3.StringUtils

@Parcelize
class Image(
    var id: String,
    var type: String? = null,
    var width: Int = 0,
    var height: Int = 0,
    var url: String,
    var urlPatternWidth: String? = null,
    var urlPatternWidthHeight: String? = null,
    var path: String? = null,
    var mimeType: String? = null
) : Parcelable {

    companion object {
        fun fake(): Image {
            return Image(
                id = System.currentTimeMillis().toString(),
                type = System.currentTimeMillis().toString(),
                url = Debug.images.random(),
                width = Dimens.purchaseableItemWidth,
                height = Dimens.purchaseableItemHeight
            )
        }

        fun init(imageCommonDto: ImageCommonDto): Image {
            return Image(
                id = imageCommonDto.imageId,
                type = imageCommonDto.type,
                width = imageCommonDto.width,
                height = imageCommonDto.height,
                url = Debug.images.random()
            )
        }

        fun init(commonImage: CommonImage): Image {
            return Image(
                id = commonImage.id,
                type = "",
                width = commonImage.width,
                height = commonImage.height,
                url = Debug.images.random()
            )
        }
    }

    /**
     * 获取方图地址
     *
     * @param width
     * @return
     */
    fun getUrlSquare(width: Int): String {
        return if (StringUtils.isNotBlank(urlPatternWidth)) {
            replacePattern(urlPatternWidth, width.toString() + "")
        } else ImageUtil.getImageUrl(id, width, ImageUtil.CropMode.ScaleCenterCrop)
    }

    private fun replacePattern(target: String?, vararg args: Any): String {
        var target = target
        target = StringUtils.replace(target, "%@", "%s")
        return String.format(target, *args)
    }
}